api/src/pcapi/routes/native/v1/account.py

```diff
@@ -67,6 +69,59 @@ def reset_recredit_amount_to_show(user: User) -> None:
     return serializers.UserProfileResponse.from_orm(user)
 
 
+@blueprint.native_v1.route("/profile/update_email", methods=["POST"])
+@spectree_serialize(
+    response_model=serializers.UserProfileResponse,
+    on_success_status=200,
+    api=blueprint.api,
+)  # type: ignore
+@authenticated_user_required
+def update_user_email(user: User, body: serializers.UserProfileEmailUpdate) -> serializers.UserProfileResponse:
+    try:
+        email_api.request_email_update(user, body.email, body.password)
+        return serializers.UserProfileResponse.from_orm(user)
+    except exceptions.EmailUpdateTokenExists:
+        raise ApiErrors(
+            {"code": "TOKEN_EXISTS", "message": "Un token actif existe déjà"},
+            status_code=400,
+        )
+    except exceptions.EmailUpdateInvalidPassword:
+        raise ApiErrors(
+            {"code": "INVALID_PASSWORD", "message": "Mot de passe invalide"},
+            status_code=400,
+        )
+    except (exceptions.EmailExistsError, exceptions.InvalidEmailError):
+        raise ApiErrors(
+            {"code": "INVALID_EMAIL", "message": "Adresse email invalide"},
+            status_code=400,
+        )
+    except exceptions.EmailUpdateLimitReached:
+        raise ApiErrors(
+            {"code": "EMAIL_UPDATE_ATTEMPTS_LIMIT", "message": "Trop de tentatives"},
+            status_code=400,
+        )
+
+
+@blueprint.native_v1.route("/profile/validate-email", methods=["PUT"])
+@spectree_serialize(on_success_status=204, api=blueprint.api)
+def validate_user_email(body: beneficiaries_serialization.ChangeBeneficiaryEmailBody) -> None:
+    try:
+        payload = beneficiaries_serialization.ChangeEmailTokenContent.from_token(body.token)
+        api.change_user_email(current_email=payload.current_email, new_email=payload.new_email)
```

Comment:
Je rajouterais bien une petite vérification (ceinture+bretelles) que le `current_email` du JWT correspond bien à l'adresse mail actuelle de l'utilisateur qui fait la requête :

```suggestion
@authenticated_user_required
def validate_user_email(user, body: beneficiaries_serialization.ChangeBeneficiaryEmailBody) -> None:
    try:
        payload = beneficiaries_serialization.ChangeEmailTokenContent.from_token(body.token)
        if user.email != payload.current_email:
            raise ValueError(
                "User tried to validate e-mail change of someone else",
                extra={
                    "user": user.id,
                    "current_email": user.email,
                    "requesting_change_for_email": payload.current_email,
                }
            )
        api.change_user_email(current_email=payload.current_email, new_email=payload.new_email)
```

Reply:
Dans une prochaine PR du coup.